### PR TITLE
Added events on CSO

### PR DESCRIPTION
**Pull request by ghanshyam-11**
Signed-off-by: ghanshyam-11 <ghanshyam.sharma1@ibm.com>

## Pull request checklist

```release-notes

```

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature Enhancement
- [ ] Test Automation 
- [ ] Code Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Community Operator listing 
- [ ] Other (please describe): 

## What is the current behavior?

-

## What is the new behavior?

-

## How risky is this change?
- [ ] Small, isolated change
- [ ] Medium, requires regression testing
- [ ] Large, requires functional and regression testing

### CODE REVIEW

File: operator/controllers/config/resources.go [hunk 1]
```diff
@@ -79,3 +79,10 @@ func GetNameForResource(name ResourceName, driverName string) string {
 		return fmt.Sprintf("%s-%s", driverName, name)
 	}
 }
+
+type ResourceKind string
+
+const (
+	Secret    ResourceKind = "Secret"
```
Comment (amdabhad) [inline operator/controllers/config/resources.go:86]: Please add this comment at the end of Secret constant definition to skip checking this in gosec scan:
// #nosec G101 false positive

Comment (ghanshyam-11) [reply operator/controllers/config/resources.go]: Modified accordingly!

File: operator/api/v1/csiscaleoperator_types.go [hunk 2]
```diff
@@ -396,11 +396,13 @@ func init() {
...
 
 const (
-	CSIConfigured CSIReason = "Successful"
-	CreateError   CSIReason = "CreateError"
-	ReadError     CSIReason = "ReadError"
-	UpdateError   CSIReason = "UpdateError"
-	DeleteError   CSIReason = "DeleteError"
-	Unknown       CSIReason = "Unknown"
-	ConfigError   CSIReason = "ConfigError"
+	//Expected when all CSI resources have been create/updated successfully
```
Comment (amdabhad) [inline operator/api/v1/csiscaleoperator_types.go:399]: create --> created

File: operator/api/v1/csiscaleoperator_types.go [hunk 3]
```diff
@@ -396,14 +396,13 @@ func init() {
...
-	ResourceNotFoundError CSIReason = "resourceNotFound"
-	ResourceConfigError   CSIReason = "configError"
+	//Expected when all CSI resources have been create/updated successfully
+	Configured CSIReason = "Configured"
+	//Expected when any of the Get/Create/Update/Delete operation fails for any k8s resource required for CSI
+	ConfigError CSIReason = "ConfigError"
+	//Initial state
+	Unknown CSIReason = "Unknown"
+	//Expected when any scale operation such as create fileset, get mount details, GUI password etc fails
+	//and when validation of driver manifest parameters fail
```
Comment (amdabhad) [inline operator/api/v1/csiscaleoperator_types.go:406]: fail --> fails

File: operator/api/v1/csiscaleoperator_types.go (hunk 2 above)
Comment (ghanshyam-11) [reply operator/api/v1/csiscaleoperator_types.go]: corrected!

**Comment by ghanshyam-11** [comment]
will raise a new PR to IBM:dev

**Pull request closed without merging**